src/App.tsx

```diff
@@ -15,36 +19,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortField {
+  All = '',
+  Alphabet = 'alphabet',
+  Length = 'length',
+} 
+
+interface SortOptions {
+  sortField: SortField;
+  isReverse: boolean;
+};
+
+function getSortedGoods(goods: Goods, { sortField, isReverse }: SortOptions) {
+  const sortedGoods = [...goods];
+
+  if (sortField) {
+    sortedGoods.sort((a, b) => {
+      switch (sortField) {
+        case SortField.Alphabet:
+          return a.localeCompare(b);
+
+        case SortField.Length:
+          return a.length - b.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverse) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = React.useState<SortField>(SortField.All);
+  const [isReverse, setIsReverse] = React.useState(false);
+
+  const sortedGoods: Goods = getSortedGoods(goodsFromServer, {
+    sortField,
+    isReverse,
+  });
+  const isChanged: boolean = isReverse || sortField !== '';
+
+  const resetOnClick = () => {
```

Comment:
```suggestion
  const handleReset = () => {
```
In accordance with convention handlers should start from work `handle`
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364